go-controller/pkg/ovn/secondary_layer2_network_controller.go

```diff
@@ -60,10 +61,20 @@ func NewSecondaryLayer2NetworkController(cnci *CommonNetworkControllerInfo, netI
 	}
 
 	if oc.allocatesPodAnnotation() {
-		podAnnotationAllocator := pod.NewPodAnnotationAllocator(
+		var podAnnotationAllocator pod.AnnotationAllocator = pod.NewPodAnnotationAllocator(
 			netInfo,
 			cnci.watchFactory.PodCoreInformer().Lister(),
 			cnci.kube)
+		if netInfo.AllowsPersistentIPs() && util.DoesNetworkRequireIPAM(netInfo) {
```

Comment:
nit: we can join these two checks in one method just as we did in cluster manager side